UserInterface/GameManager/GameManager.cs

```diff
@@ -69,43 +90,85 @@ private void RunGame()
                 Console.CursorVisible = false;
                 Pair? currentPair = _pairManager.Pair;
 
-                _board.FillBoardWithAnimals(_animals);
+                _board.FillBoardWithObjects(_gameObjects);
 
                 _userInterface.DrawBoard(_board);
-                _board.RemoveAnimalFromBoard(_animals);
-
-                _animalActions.Move(_board, _animals);
+                _board.RemoveGameObjectFromBoard(_gameObjects);
+                
+                _movementManager.Act(_gameObjects, _board);
 
                 _pairManager.RemoveNonExistsPairs(_pairs);
 
-                _pairManager.CheckPairForExistence(_pairs, _board, _animals, _children);
+                _pairManager.CheckPairForExistence(_pairs, _board, _gameObjects, _childrenGameObjects);
 
-                _pairManager.CreatePair(_animals);
+                _pairManager.CreatePair(_gameObjects);
 
                 if (currentPair != null)
                 {
                     _pairs.Add(currentPair);
                 }
 
-                _animalActions.AddChildren(_animals, _children);
-                _children.Clear();
+                _objectManager.AddChildrenObject(_gameObjects, _childrenGameObjects);
+                _childrenGameObjects.Clear();
```

Comment:
This looks low level code. Probably better to put it in 
 _objectManager.AddChildrenObject()
 method.

Reply:
Done.